Source/WebKit/GPUProcess/media/InitializationSegmentInfo.serialization.in

```diff
@@ -31,7 +31,7 @@ struct WebKit::InitializationSegmentInfo {
 
 [Nested] struct WebKit::InitializationSegmentInfo::TrackInformation {
     WebKit::MediaDescriptionInfo description;
-    WebKit::TrackPrivateRemoteIdentifier identifier;
+    uint64_t id;
```

Comment:
We shouldn't be passing non-strongly typed identifiers via IPC. This change seems like a regression security wise?
Ids sent over IPC need to be validated because they're often used to look up in HashMaps and such and lead to crashes on recipient side.

Reply:
There's heaps of places where we are passing int across though when it comes to selected track, track index, number of frames. This made it consistent.

To me if we want to assess that the int is a valid one, it's the wrong place to start.

Here, every path is tested to check if we have a new track and it will safely abort

Reply:
additionally, what we want to pass is the track ID as is found in the original container, which is an int. Storing that int as an identifier, pass it over IPC only to read it as an int again (after looking it up in a HashMap) seems wasteful.

> they're often used to look up in HashMaps and such and lead to crashes on recipient side.
Looking up the TrackPrivateRemoteIdentifier in a map to find the original ID is exactly what it was doing before, so we removed possibility of mis-used, not add some.

Reply:
The TrackPrivateRemoteIdentifier that was received over IPC was already validated at the IPC layer. If it were an identifier that could not be looked up in a HashMap, IPC decoding would have failed. So what you call wasteful is actually valuable security, which we worked to achieve.

Now that you're passing an int, we no longer get ANY IPC validation. This is not acceptable. I think this patch is a step in the wrong direction.

Reply:
I provided further details in the commit log as to the decision of no longer using HashMap and how there is nothing to validate as the ID is never invalid. Please reconsider.

Reply:
I removed my objection given that I don't have a better proposal at the moment given the id requirements. @achristensen07 may have an idea though.

Reply:
I added a test with a file that is using 2^64-1 as track id